test/Dapr.Client.Test/InvokeApiTest.cs

```diff
@@ -274,11 +292,56 @@ public async Task InvokeMethodAsync_WithReturnTypeAndData_UsesConfiguredJsonSeri
             httpClient.Requests.TryDequeue(out var entry).Should().BeTrue();
-            var envelope = await GrpcUtils.GetEnvelopeFromRequestMessageAsync<InvokeServiceEnvelope>(entry.Request);
+            var envelope = await GrpcUtils.GetEnvelopeFromRequestMessageAsync<InvokeServiceRequest>(entry.Request);
             envelope.Id.Should().Be("test");
-            envelope.Method.Should().Be("test");
-            var json = envelope.Data.Value.ToStringUtf8();
-            json.Should().Be(JsonSerializer.Serialize(invokeRequest, jsonOptions));
+            envelope.Message.Method.Should().Be("test");
+
+            DataWithContentType data = envelope.Message.Data.Unpack<DataWithContentType>();
+            string s = data.Body.ToStringUtf8();
+
+            var t = JsonSerializer.Serialize(invokeRequest, jsonOptions);
+            t.Should().Be(s);
+
+            SendResponse(invokedResponse, entry, jsonOptions);
+            var response = await task;
+
+            response.Name.Should().Be(invokedResponse.Name);
+        }
+
+        [Fact]
+        public async Task InvokeMethodAsync_WithReturnTypeAndData_WithNonDefaultVerb_WithQueryString_UsesConfiguredJsonSerializerOptions()
+        {
+            // Configure Client
+            var httpClient = new TestHttpClient();
+            var jsonOptions = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+            var daprClient = new DaprClientBuilder()
+                .UseGrpcChannelOptions(new GrpcChannelOptions { HttpClient = httpClient })
+                .UseJsonSerializationOptions(jsonOptions)
+                .Build();
+
+            var invokeRequest = new InvokeRequest() { RequestParameter = "Hello " };
+            var invokedResponse = new InvokedResponse { Name = "Look, I was invoked!" };
+
+            Dictionary<string, string> queryString = new Dictionary<string, string>();
+            queryString.Add("key1", "value1");
+            var task = daprClient.InvokeMethodAsync<InvokeRequest, InvokedResponse>("test", "test", invokeRequest, HTTPVerb.Put, queryString);
+
+            // Get Request and validate            
+            httpClient.Requests.TryDequeue(out var entry).Should().BeTrue();
+            var envelope = await GrpcUtils.GetEnvelopeFromRequestMessageAsync<InvokeServiceRequest>(entry.Request);
+            envelope.Id.Should().Be("test");
+            envelope.Message.Method.Should().Be("test");
+            envelope.Message.HttpExtension.Verb.Should().Be(HTTPExtension.Types.Verb.Put);
+            envelope.Message.HttpExtension.Querystring.Count.Should().Be(1);
+            envelope.Message.HttpExtension.Querystring.ContainsKey("key1").Should().BeTrue();
+            envelope.Message.HttpExtension.Querystring["key1"].Should().Be("value1");
+
+
+            DataWithContentType data = envelope.Message.Data.Unpack<DataWithContentType>();
+            string s = data.Body.ToStringUtf8();
```

Comment:
Please use descriptive variable names rather than s and t.
Be consistent with usage of var.